mysys/system_key.cc

```diff
@@ -64,7 +64,7 @@ uchar *parse_system_key(const uchar *key, const size_t key_length,
 
   if (key == nullptr || key_length == 0) return nullptr;
 
-  for (; key_version_length < key_length && key[key_version_length] != ':';
+  for (; key[key_version_length] != ':' && key_version_length < key_length;
```

Comment:
Any reason for swapping these conditions?

Reply:
It was part of one of the big keyring encryption commits. But you are right, the current version is better, I'll revert this change.